src/sinks/file/mod.rs

```diff
@@ -53,6 +60,46 @@ impl Default for Encoding {
     }
 }
 
+#[derive(Deserialize, Serialize, Debug, Eq, PartialEq, Clone, Copy)]
+#[serde(rename_all = "snake_case")]
+pub enum Compression {
+    Gzip,
+    None,
+}
```

Comment:
Note that there already exists an `enum Compression` in `src/sinks/util/buffer/mod.rs`, which matches this definition almost exactly. Would that be usable here to avoid the duplicate definition?

Reply:
I had noticed that. It is almost similar, but I felt that implementation more tied to flate2/std::io::Write. 
While this one relies on async_compression and async writes via tokio. With async_compression, it can be quickly extended for other compression types if required.